metricflow/specs/where_filter_entity.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import List, Sequence
+
+from dbt_semantic_interfaces.call_parameter_sets import (
+    EntityCallParameterSet,
+    FilterCallParameterSets,
+)
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.references import EntityReference
+from typing_extensions import override
+
+from metricflow.specs.column_assoc import ColumnAssociationResolver
+from metricflow.specs.query_interface import QueryInterfaceDimension, QueryInterfaceEntityFactory
+from metricflow.specs.specs import EntitySpec
+
+
+class WhereFilterEntity(ProtocolHint[QueryInterfaceDimension]):
+    """An entity that is passed in through the where filter parameter."""
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceDimension:
+        return self
+
+    def __init__(self, column_name: str):  # noqa
+        self.column_name = column_name
+
+    def grain(self, _grain: str) -> WhereFilterEntity:
+        """The time granularity."""
+        raise NotImplementedError
+
+    def alias(self, _alias: str) -> WhereFilterEntity:
+        """Renaming the column."""
+        raise NotImplementedError
+
+    def __str__(self) -> str:
+        """Returns the column name.
+
+        Important in the Jinja sandbox.
+        """
+        return self.column_name
+
+
+class WhereFilterEntityFactory(ProtocolHint[QueryInterfaceEntityFactory]):
+    """Creates a WhereFilterDimension."""
```

Comment:
nit:
```suggestion
    """Creates a WhereFilterEntity."""
```

Reply:
Thank you! Must have been a copy-paste error